portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,33 +29,41 @@
 import java.util.ArrayList; // import the ArrayList class
 import com.google.gson.Gson;
 import java.util.List;
+import com.google.appengine.api.datastore.FetchOptions;
 
 
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Does this initialization fail at this point? It's not a bad idea to do this once early on and then all the functions can share it.

If it does fail, a common theme is to have like a getDatastore() function that checks if datastore == null and sets datastore to DatastoreServiceFactory.getDatastoreService(). Then always returns datastore. This is a pattern you'll see very often in code.